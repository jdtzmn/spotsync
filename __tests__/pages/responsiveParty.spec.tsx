import * as React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import ResponsiveParty from 'pages/responsiveParty'

describe('ResponsiveParty', () => {
  let wrapper: ShallowWrapper
  beforeEach(() => {
    wrapper = shallow(<ResponsiveParty />)
  })

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should set isMobile to true when less than 768px', () => {
    // Change the viewport to 767px
    Object.defineProperty(window, 'innerWidth', {
      writable: true,
      value: 767
    })
    window.dispatchEvent(new Event('resize'))

    expect(wrapper.state('isMobile')).toBe(true)
  })

  it('should set isMobile to false when wider than 768px', () => {
    // Change the viewport to 1024px
    Object.defineProperty(window, 'innerWidth', {
      writable: true,
      value: 1024
    })
    window.dispatchEvent(new Event('resize'))

    expect(wrapper.state('isMobile')).toBe(false)
  })

  it('should render MobileParty when less than 768px', () => {
    // Change the viewport to 767px
    Object.defineProperty(window, 'innerWidth', {
      writable: true,
      value: 767
    })
    window.dispatchEvent(new Event('resize'))

    expect(wrapper.find('MobileParty')).toHaveLength(1)
    expect(wrapper.find('DesktopParty')).toHaveLength(0)
  })

  it('should render DesktopParty when wider than 768px', () => {
    // Change the viewport to 1024px
    Object.defineProperty(window, 'innerWidth', {
      writable: true,
      value: 1024
    })
    window.dispatchEvent(new Event('resize'))

    expect(wrapper.find('DesktopParty')).toHaveLength(1)
    expect(wrapper.find('MobileParty')).toHaveLength(0)
  })
})
