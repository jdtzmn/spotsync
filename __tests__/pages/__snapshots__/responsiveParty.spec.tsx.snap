// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResponsiveParty should render correctly 1`] = `
<main>
  <DesktopParty />
</main>
`;
